components/apps.js

```diff
@@ -649,33 +647,82 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
  * Check if this account owns a depot. Only works if enablePicsCache option is enabled and appOwnershipCached event
  * has been emitted.
  * @param {int} depotid
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * @param {object} options - Options for what counts for ownership
  * @returns {boolean}
  */
-SteamUser.prototype.ownsDepot = function(depotid, excludeSharedLicenses) {
-	return this.getOwnedDepots(excludeSharedLicenses).indexOf(parseInt(depotid, 10)) != -1;
+SteamUser.prototype.ownsDepot = function(depotid, options) {
+	return this.getOwnedDepots(options).indexOf(parseInt(depotid, 10)) != -1;
 };
 
 /**
- * Returns an array of package IDs this account owns. Only works if enablePicsCache option is enabled and appOwnershipCached event
- * has been emitted.
- * @param {boolean} [excludeSharedLicenses=false] - Pass true to exclude licenses that we have through family sharing
+ * Returns an array of package IDs this account owns (different from owned licenses). Only works if enablePicsCache
+ * option is enabled and appOwnershipCached event has been emitted.
+ * @param {object} options - Options for what counts for ownership
  * @returns {int[]}
  */
-SteamUser.prototype.getOwnedPackages = function(excludeSharedLicenses) {
+SteamUser.prototype.getOwnedPackages = function(options) {
 	if (this.steamID.type != SteamID.Type.ANON_USER && !this.licenses) {
 		throw new Error("We don't have our license list yet.");
 	}
 
+	// We're anonymous
 	if (this.steamID.type == SteamID.Type.ANON_USER) {
 		return [17906];
 	}
 
 	// We're an individual user
-	let packages = this.licenses;
-	if (excludeSharedLicenses) {
-		packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+	this._ensurePicsCache();
+
+	// Adds support for the previous syntax
+	if (typeof options === 'boolean') {
+		options = {
+			"shared": !options // !excludeSharedLicenses
+		}
 	}
+
+	const defaults = {
+		shared: false, // By default, exclude shared licenses
+		expiring: false // By default, exclude licenses that are going to expire (free weekends)
+	};
+	options = { ...defaults, ...options };
+
+	let packages = this.licenses;
+	packages = packages.filter((license) => {
```

Comment:
I hope this filter function is clear enough for you. It was a bit of a headache to get it right and clear. Having free promotions (yours to keep) have the `expirytime` property made it difficult.